tensorpack/dataflow/format.py

```diff
@@ -103,6 +118,8 @@ def get_data(self):
                 if k != '__keys__':
                     yield [k, v]
         else:
+            if not hasattr(self, "rng"):
+                self.reset_state()
```

Comment:
The protocol is that a user always have to call `reset_state()` before `get_data` is first called.